File: sdkwork-ai-sdk-typescript/src/api/rerank.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { RerankRequest, RerankResponse } from '../types';


export class RerankApi2 {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Rerank documents */
  async rerank(body: RerankRequest): Promise<RerankResponse> {
    return this.client.post<RerankResponse>(aiApiPath(`/v1/rerank`), body);
  }
}

export function createRerankApi2(client: HttpClient): RerankApi2 {
  return new RerankApi2(client);
}

File: sdkwork-ai-sdk-go/api/rerank.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type RerankApi struct {
    client *sdkhttp.Client
}

func NewRerankApi(client *sdkhttp.Client) *RerankApi {
    return &RerankApi{client: client}
}

// Rerank documents
func (a *RerankApi) Rerank(body sdktypes.RerankRequest) (sdktypes.RerankResponse, error) {
    raw, err := a.client.Post(AiApiPath("/rerank"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.RerankResponse
        return zero, err
    }
    return decodeResult[sdktypes.RerankResponse](raw)
}
